thunder/torch/__init__.py

```diff
@@ -3489,6 +3489,21 @@ def embedding_backward(grad, indices, num_weights, padding_idx, scale_grad_by_fr
     return result
 
 
+@torchsymbol(torch.nn.functional.one_hot, id="torch.nn.functional.one_hot", is_method=False)
+def one_hot(a: TensorProxy, /, num_classes: int) -> TensorProxy:  # Should we use TensorProxy for a?
+    if num_classes == -1:
+        # TODO: implement when we're ready to support auto inference using `.item()`
+        utils.check(False, lambda: "num_classes=-1 not supported yet.", exception_type=NotImplementedError)
+    # TODO: should we implement this check? I suspect this might hinder performance, also we do not support .any)
+    #  utils.check(a.any() >= 0, lambda f"input tensor should have non-negative values", exception_type=ValueError)
+
+    canvas = zeros(*a.shape, num_classes, dtype=dtypes.int64)  # this is equivalent to the use of .long in torch
+    index = a.unsqueeze(-1)
+    src = ones_like(index, dtype=dtypes.int64)
```

Comment:
The construction of `src` avoids the above issue but requires unsqueezing `a` just to use that result as an example. I suggest mimicking the above approach and calling `ones`, and computing the desired shape as `a.shape + (1,)`.